### PR TITLE
jsdoc: lib/api/readable.js, fix some types

### PR DESCRIPTION
**Pull request by Uzlopak**
Adds jsdoch to readable, fixes some aspects which i will comment

### CODE REVIEW

File: lib/api/readable.js [hunk 1]
```diff
@@ -174,26 +273,34 @@ class BodyReadable extends Readable {
...
+            signal.removeEventListener('abort', onAbort)
+            if (signal.aborted) {
+              reject(signal.reason ?? new AbortError())
+            } else {
+              resolve(null)
+            }
+          })
+      } else {
+        this.on('close', resolve)
       }
```
Comment (Uzlopak) [inline lib/api/readable.js:299]: reorderering so that we only do signal specific stuff like creating onAbort if needed

File: lib/api/readable.js [hunk 2]
```diff
@@ -158,14 +248,23 @@ class BodyReadable extends Readable {
...
   async dump (opts) {
     const signal = opts?.signal
 
     if (signal != null && (typeof signal !== 'object' || !('aborted' in signal))) {
       throw new InvalidArgumentError('signal must be an AbortSignal')
     }
 
-    const limit = Number.isFinite(opts?.limit) ? opts.limit : 128 * 1024
+    const limit = opts?.limit && Number.isFinite(opts.limit)
+      ? Math.trunc(opts.limit)
```
Comment (Uzlopak) [inline lib/api/readable.js:266]: use trunc to ensure it is an integer

**Review by ronag — APPROVED** [review]

File: lib/api/readable.js [hunk 3]
```diff
@@ -36,12 +49,19 @@ class BodyReadable extends Readable {
...
     this[kConsume] = null
     this[kBytesRead] = 0
+    /**
+     * @type {ReadableStream|null}
+     */
     this[kBody] = null
     this[kUsed] = false
     this[kContentType] = contentType
-    this[kContentLength] = Number.isFinite(contentLength) ? contentLength : null
+    this[kContentLength] = Number.isFinite(contentLength) ? Math.trunc(contentLength) : null
```
Comment (Uzlopak) [inline lib/api/readable.js:64]: use trunc to ensure an integer

Comment (Uzlopak) [reply lib/api/readable.js]: on second thought this is not necessary

**Comment by Uzlopak** [comment]
@ronag 
PTAL

**Review by ronag — APPROVED** [review]

**Pull request merged**